src/views/forms/TradeForm/TargetLeverageInput.tsx

```diff
@@ -106,9 +106,10 @@ export const TargetLeverageInput = () => {
           type={InputType.Leverage}
           value={leverage}
           max={maxLeverage}
-          onChange={({ floatValue }: NumberFormatValues) =>
-            setLeverage(floatValue?.toString() ?? '')
-          }
+          onChange={({ floatValue }: NumberFormatValues) => {
+            setLeverage(floatValue?.toString() ?? '');
+            debouncedSetAbacusLeverage(floatValue ?? '');
```

Comment:
The `useState` is purely for rendering purposes?

Reply:
wondering if there is a way to leverage the `useDebounce` hook for this

Reply:
Yeah in retrospect we should just refactor this because it's a bit of a mess, but let's merge for now .